src/node/internal/internal_assert.ts

```diff
@@ -374,6 +374,213 @@ export function notDeepStrictEqual(
+      for (let i = 0; i < expected.length; i++) {
+        if (i in expected) {
+          if (
+            !(i in actual) ||
+            !isPartialDeepStrictEqual(actual[i], expected[i], memo)
+          ) {
+            return false;
+          }
+        }
+      }
+      return true;
+    }
+
+    // Handle Sets
+    if (expected instanceof Set) {
+      if (!(actual instanceof Set)) {
+        return false;
+      }
+
+      // Check if all expected values are in actual set
+      for (const expectedValue of expected) {
+        let found = false;
+        for (const actualValue of actual) {
+          if (isDeepStrictEqual(expectedValue, actualValue)) {
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+    // Handle Maps
+    if (expected instanceof Map) {
+      if (!(actual instanceof Map)) {
+        return false;
+      }
+
+      // Check if all expected key-value pairs are in actual map
+      for (const [expectedKey, expectedValue] of expected) {
+        let found = false;
+        for (const [actualKey, actualValue] of actual) {
+          if (isDeepStrictEqual(expectedKey, actualKey)) {
+            if (!isPartialDeepStrictEqual(actualValue, expectedValue, memo)) {
+              return false;
+            }
+            found = true;
+            break;
+          }
+        }
+        if (!found) {
+          return false;
+        }
+      }
+      return true;
+    }
+
+    // Check prototypes are the same (required for deepStrictEqual behavior)
+    if (Object.getPrototypeOf(actual) !== Object.getPrototypeOf(expected)) {
+      return false;
+    }
+
+    // Handle regular objects
+    const expectedKeys = Object.keys(expected);
+
+    for (const key of expectedKeys) {
+      if (!(key in actual)) {
+        return false;
+      }
+
+      if (
+        !isPartialDeepStrictEqual(
+          (actual as any)[key],
+          (expected as any)[key],
+          memo
+        )
+      ) {
+        return false;
+      }
+    }
+
+    // Handle symbol properties
+    const expectedSymbols = Object.getOwnPropertySymbols(expected);
+    for (const symbol of expectedSymbols) {
+      if (expected.propertyIsEnumerable(symbol)) {
+        if (
+          !(symbol in actual) ||
+          !isPartialDeepStrictEqual(
+            (actual as any)[symbol],
+            (expected as any)[symbol],
+            memo
+          )
+        ) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  } finally {
+    // Clean up memo to prevent memory leaks
+    if (memo.has(actual)) {
+      const actualSet = memo.get(actual)!;
+      actualSet.delete(expected);
+      if (actualSet.size === 0) {
+        memo.delete(actual);
+      }
+    }
+  }
+}
+
+export function partialDeepStrictEqual(
+  actual: unknown,
+  expected: unknown,
+  message?: string | Error
+) {
```

Comment:
How does the build/tests pass without return value type?

Reply:
<img width="400" height="300" alt="image" src="https://github.com/user-attachments/assets/977f70ca-3b2d-4c81-b059-a314f7e5d2e7" />

Reply:
Line 28 :-(